# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/ViewportUtils.h
# Commit: 4312b2b5dda8
# Full Hash: 4312b2b5dda8c1af7b56ac1c55ab762aff485a2c
# Author: Botond Ballo <botond@mozilla.com>
# Date: 2020-04-28 16:30:11
# Regressor Bug: 1556556
# File Overlap Count: 1
# Description:
#   Bug 1556556 - Move GetCallbackTransform() into a new ViewportUtils class. r=kats
#   
#   This function (and helper functions that wrap it) will be used extensively
#   throughout layout code, so keeping it in APZCCallbackHelper seems awkward.
#   
# ==============================================================================

diff -r d11dbf6403a5 -r 4312b2b5dda8 layout/base/ViewportUtils.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/layout/base/ViewportUtils.h	Tue Apr 28 01:35:05 2020 +0000
@@ -0,0 +1,36 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_ViewportUtils_h
+#define mozilla_ViewportUtils_h
+
+#include "Units.h"
+#include "mozilla/layers/ScrollableLayerGuid.h"
+
+namespace mozilla {
+
+class ViewportUtils {
+ public:
+  /* Return a "callback transform" to be applied to the coordinates of input
+     events targeting content inside the scroll frame identified by |aScrollId|.
+     The callback transform has two components:
+     1. The pres shell resolution, representing the pinch-zoom scale
+        (if the scroll frame |aScrollId| is inside the resolution, which
+        is most of the time).
+     2. A translation representing async scrolling. This can contain:
+         - For any scroll frame, a scroll component resulting from the main
+           thread incompletely applying an APZ-requested scroll position.
+         - For the RCD-RSF only, a persistent component representing the
+           offset of the visual viewport relative to the layout viewport.
+       The translation is accumulated for all scroll frames form |aScrollId|
+       up to the root, using values populated in
+       APZCCallbackHelper::UpdateCallbackTransform. See that method's
+       documentation for additional details. */
+  static CSSToCSSMatrix4x4 GetCallbackTransform(
+      layers::ScrollableLayerGuid::ViewID aScrollId);
+};
+
+}  // namespace mozilla
+
+#endif /* mozilla_ViewportUtils_h */